Sources/libhostmgr/Foundation+Extensions.swift

```diff
@@ -13,6 +13,29 @@ extension ProcessInfo {
         let identifier = String(bytes: data, encoding: .ascii)!.trimmingCharacters(in: .controlCharacters)
         return ProcessorArchitecture(rawValue: identifier)!
     }
+
+    public var physicalProcessorCount: Int {
+        let output = Pipe()
+
+        do {
+            let task = Process()
+            task.launchPath = "/usr/sbin/sysctl"
+            task.arguments = ["-n", "hw.physicalcpu"]
+            task.standardOutput = output
+            try task.run()
+
+            let cpuCountData = output.fileHandleForReading.readDataToEndOfFile()
+            let cpuCountString = String(
+                data: cpuCountData,
+                encoding: .utf8
+            )!.trimmingCharacters(in: .whitespacesAndNewlines)
```

Comment:
What do you think using `TSCBasic.Process` here ([an example](https://github.com/apple/swift-tools-support-core/blob/e915be3531f656285292c8506eb477185902b91e/Tests/TSCBasicTests/ProcessTests.swift#L24-L27))?

Reply:
Ah this is a great reminder – I've been meaning to move this over to a syscall. This is done in https://github.com/Automattic/hostmgr/pull/46/commits/109ed679830bd0e215608082536d5159b62b4804, and we use `TSCBasic.Process` for the test – this gives a totally separate implementation to validate against.

WDYT?